compiler/packagehandling.nim

```diff
@@ -39,11 +39,11 @@ proc getPackageName*(conf: ConfigRef; path: string): string =
     if parents <= 0: break
 
 proc fakePackageName*(conf: ConfigRef; path: AbsoluteFile): string =
-  # foo/../bar becomes foo@..@bar
-  result = relativeTo(path, conf.projectPath, '/').string.multiReplace({"/": "@", "@": "@@"})
+  # foo-#head/../bar becomes @foo-@hhead@s..@sbar
+  result = "@" & relativeTo(path, conf.projectPath, '/').string.multiReplace({"/": "@s", "#": "@h", "@": "@@"})
```

Comment:
Gah, why a preceeding '@'? It's already bad enough.

Reply:
Yeah, that's to avoid leading dot. Anyway, this is just a quick and somewhat dirty fix. Feel free to close it in favor of a better approach (prefix shortening?).

Reply:
Some other programming lang probably is doing this somehow somewhere else.

Ive seen PowerShell uses base64 to avoid problems with paths/quoting.

Reply:
> Ive seen PowerShell uses base64 to avoid problems with paths/quoting.

@Araq, see? It can be even uglier :)